imports/client/components/PuzzlePage.jsx

```diff
@@ -701,58 +699,54 @@ class PuzzlePageMetadata extends React.Component {
           tags={this.props.allTags}
           onSubmit={this.onEdit}
         />
-        <div>
-          <div className="puzzle-metadata-row">
-            <div className="puzzle-metadata-right">
-              {this.props.puzzle.url && (
-                <a
-                  className="puzzle-metadata-external-link-button"
-                  href={this.props.puzzle.url}
-                  target="_blank"
-                  rel="noreferrer noopener"
-                >
-                  Puzzle
-                </a>
-              )}
-            </div>
-            <div className="puzzle-metadata-left">
-              {this.editButton()}
-              {' '}
-              {this.props.isDesktop &&
-                <span className="puzzle-metadata-title">{this.props.puzzle.title}</span>}
-              {!this.props.isDesktop && this.props.document &&
-                <DocumentDisplay document={this.props.document} displayMode="link" />}
-              {' '}
-              {this.props.puzzle.answer && answerComponent}
-              {' '}
-              {!hideViewCount && (
-                <ViewCountDisplayContainer
-                  count={this.props.viewCount}
-                  name={`puzzle:${this.props.puzzle._id}`}
-                />
-              )}
-            </div>
-          </div>
-          <div className={classnames('puzzle-metadata-row', this.props.isDesktop && 'puzzle-metadata-tag-editor-row')}>
-            {!isAdministrivia && (
-              <div className="puzzle-metadata-right">
-                <Button className="puzzle-metadata-guess-button" onClick={this.showGuessModal}>
-                  {this.props.puzzle.answer ? `View ${guessesString}` : `Submit answer (${guessesString})`}
-                </Button>
-              </div>
-            )}
-            <div className="puzzle-metadata-left">
-              <span className="puzzle-metadata-tags">Tags:</span>
-              <TagList
-                puzzleId={this.props.puzzle._id}
-                tags={tags}
-                onCreateTag={this.onCreateTag}
-                onRemoveTag={this.onRemoveTag}
-                linkToSearch={false}
-                showControls={this.props.isDesktop}
+        <div className="puzzle-metadata-row">
+          <div className="puzzle-metadata-title-set">
+            {this.editButton()}
+            {' '}
+            <span className="puzzle-metadata-title">{this.props.puzzle.title}</span>
+            {' '}
+            {!hideViewCount && (
+              <ViewCountDisplayContainer
+                count={this.props.viewCount}
+                name={`puzzle:${this.props.puzzle._id}`}
               />
-            </div>
+            )}
           </div>
+          {this.props.puzzle.answer && answerComponent}
+        </div>
+        <div className={classnames('puzzle-metadata-row', this.props.isDesktop && 'puzzle-metadata-tag-editor-row')}>
+          <TagList
```

Comment:
Can I convince you to add the "Tags:" label back here?  Without it, on a puzzle that doesn't have any tags, it's not super clear what the tags are or what the + button is doing.

Reply:
That's a good point.  If your concern is mostly the empty tag set case, I'd say we should just add a "(No Tags)" message.  But I'm also not particularly opposed to adding "Tags" back.  Your call.

Reply:
I think I like adding `Tags:` back.  It gives us a way to tell people what they should call the colored blobs.  Hunts and puzzles are hinted at by the nav breadcrumbs, guesses are hinted at by the button for showing previous guesses, but tags don't have much visibility for what to call that concept other than that label.